src/transactions/feed/JumpstartTransactionDetailsScreen.tsx

```diff
@@ -0,0 +1,320 @@
+
+type Props = NativeStackScreenProps<StackParamList, Screens.JumpstartTransactionDetailsScreen>
+
+async function getClaimDataAndStatus(
+  jumpstartContractAddress: Address,
+  networkId: NetworkId,
+  transactionHash: Hash
+) {
+  const viemClient = publicClient[networkIdToNetwork[networkId]]
+
+  const transactionReceipt = await viemClient.getTransactionReceipt({
+    hash: transactionHash,
+  })
+
+  const parsedLogs = parseEventLogs({
+    abi: walletJumpstart.abi,
+    eventName: ['ERC20Deposited'],
+    logs: transactionReceipt.logs,
+  })
+
+  if (parsedLogs.length != 1) {
+    throw new Error('Unexpected number of matching logs')
+  }
+
+  const { beneficiary, index } = parsedLogs[0].args
+
+  Logger.debug('Decoded event', { beneficiary, index })
+
+  const erc20Claim = await viemClient.readContract({
+    address: jumpstartContractAddress as Address,
+    abi: walletJumpstart.abi,
+    functionName: 'erc20Claims',
+    args: [beneficiary, index],
+  })
+
+  const claimed = erc20Claim[3]
+
+  Logger.debug(`Reward deposited in ${transactionHash} ${claimed ? 'was' : 'was NOT'} claimed`)
+
+  return { beneficiary, index: Number(index), claimed }
+}
+
+function JumpstartTransactionDetailsScreen({ route }: Props) {
+  const { transaction } = route.params
+  const { t } = useTranslation()
+
+  const [reclaimTx, setReclaimTx] = useState<SerializableTransactionRequest | null>(null)
+  const transactionString = useMemo(() => (reclaimTx ? JSON.stringify(reclaimTx) : ''), [reclaimTx])
+
+  const transactionTokenInfo = useTokenInfo(transaction.amount.tokenId)
+  const parsedAmount = new BigNumber(transaction.amount.value).abs()
+  const token = useTokenInfo(transaction.amount.tokenId)
+  const isDeposit = transaction.type === TokenTransactionTypeV2.Sent
+  const jumpstartContractAddress = transaction.address
+
+  const walletAddress = useSelector(walletAddressSelector)
+  const bottomSheetRef = useRef<BottomSheetRefType>(null)
+
+  const title =
+    transaction.type === TokenTransactionTypeV2.Sent
+      ? t('feedItemJumpstartTitle')
+      : t('feedItemJumpstartReceivedSubtitle')
+
+  const fetchClaimData = useAsync(
+    async () => {
+      if (!isDeposit) {
+        return
+      }
+      return await getClaimDataAndStatus(
+        jumpstartContractAddress as Address,
+        transaction.networkId,
+        transaction.transactionHash as Hash
+      )
+    },
+    [],
+    {
+      onError: (error) => {
+        // TODO: Show error message in the UI
+        Logger.error(TAG, 'Failed to fetch escrow data', error)
+      },
+    }
+  )
+
+  if (!token) {
+    // should never happen
+    Logger.error(TAG, 'Token is undefined')
+    return null
+  }
+
+  const onReclaimPress = () => {
+    if (!fetchClaimData.result) {
+      Logger.error(TAG, 'No escrow data found when trying to reclaim')
+      return
+    }
+
+    const { beneficiary, index } = fetchClaimData.result
+
+    setReclaimTx({
```

Comment:
we should move this to the `onSuccess` handler of the `useAsync`. state changes are batched so doing it here means that the reclaimTx might not have been set to this new value when the bottom sheet appears. i don't see the advantage of doing it here

Reply:
Sorry, this is also changed in the next PR, think of it as a mock for this PR 😅 

We need the gas estimation before setting the reclaimTx, and that happens when the user taps on the "Reclaim" button. Or... Maybe we can try to estimate the gas when we fetch the escrow data even if the user doesn't tap on "Reclaim"? It might take a bit more of time to load at the beginning but then the experience will be smoother.
I'm gonna do that in the following PR since I implemented the prepared transaction there. Happy to follow this discussion there.